core/src/main/java/org/mule/runtime/core/internal/store/SharedPartitionedPersistentObjectStore.java

```diff
@@ -27,6 +32,29 @@
 
   public static final String SHARED_PERSISTENT_OBJECT_STORE_KEY = "_defaultSharedPersistentObjectStore";
 
+  /**
+   * Uses an static field to control access from different instances of this partitioned persistent object store
+   * between different deploymennts, registries.
+   */
+  private static Map<String, PersistentObjectStorePartition> partitionsByName =
+      new ConcurrentHashMap<String, PersistentObjectStorePartition>() {
+
+        @Override
+        public PersistentObjectStorePartition put(String key, PersistentObjectStorePartition value) {
+          // Creates an instance of the information to avoid referencing to the muleContext as the same persistentObjectStorePartition
+          // is used by different muleContexts
+          return super.put(key, new PersistentObjectStorePartitionData(value.getPartitionName(), value.getPartitionDirectory()));
+        }
+
+        @Override
+        public PersistentObjectStorePartition putIfAbsent(String key, PersistentObjectStorePartition value) {
```

Comment:
is it really necessary? Doesn't the ConcurrentHashMap implementation call put() when calling putIfAbsent()?

Reply:
No, it does call an internal method that I can't override but it is not delegating to the put method.